include/net/tls_credentials.h

```diff
@@ -0,0 +1,103 @@
+/*
```

Comment:
@rlubos , so this would be the commit whose message can be updated with the description of 2-stage credential management as suggested in https://github.com/zephyrproject-rtos/zephyr/pull/9007#issuecomment-406224610